nn/transformer.py

```diff
@@ -0,0 +1,231 @@
+"""
+Transformer Modules
+"""
+
+from __future__ import annotations
+from typing import Optional, Any, Union
+from . import relu, layer_norm, dropout, LayerRef, Module, ModuleList, Linear
```

Comment:
As discussed, do not do the global imports.
Instead just:
```suggestion
import . as nn
```

Reply:
My IDE does not recognize this.

Reply:
Error message is `identifier expected`

Reply:
importing `__init__` would work but this does not feel right either.

Reply:
`from .. import nn` would work

But regarding this: I would prefer import `LayerRef` at least and maybe also `Module` directly, since I feel like it makes the typing etc. harder to read with nn. prefix. Or would you not like that?

Reply:
But you get the idea. Then it's maybe `from .. import nn`.

Reply:
> it makes the typing etc. harder to read with nn. prefix

Why?

Reply:
Nvm  `from .. import nn would work` this also imports something else that is not the way we want it.

Reply:
Because with the `from ..` nn.relu is suddenly a tensorflow function. So what definitely works is `import returnn_common.nn as nn` but that would be a global import

>Why?

For me the `nn.` prefex is annoying to read, but its personal preference I guess, can also write it with the prefix.

Reply:
Hmm changed it for LayerRef to nn. locally and I guess its not as bad as I first thought. Nvm then.

So just need to agree on a proper way to import now. From my understanding you dont want global imports, I did not find a local way yet though, or at least didnt come up with the correct syntax.

Reply:
> Nvm `from .. import nn` would work this also imports something else that is not the way we want it.

I don't understand what you mean.

Reply:
>I don't understand what you mean.

The import does not import the `nn` submodule but something else (via `_lazy_loader`) which then does not find for example `nn.Module` and `nn.relu` references to a bunch of other functions which are not from `rc`.

Reply:
That sounds like a bug in `_lazy_loader`? Can't you just fix that?

Reply:
pushed the changes we accumulated this far now. I can have a try to look into it.

Reply:
So this seems to be a common problem in the interaction between IDE and `_lazy_loader`. 
I am now searching whether I can find a fix for this somewhere.

Reply:
I don't really understand.

Reply:
Okay I found the problem. The variable the lazy loader is saved in is called `nn` aswell, if we import from `..` this imports the variable instead of the module. Replacing the name by `_nn` (or some other name you prefer) will fix this.

Reply:
No, this is wrong. This is exactly the point. We want that. It should import the variable. But anyway, this does not belong here into this PR.

Reply:
I pushed some small change. Can you try whether this fixes the PyCharm issue?

Reply:
I pushed another change. Can you try again?

Reply:
yes it is working now, thank you!